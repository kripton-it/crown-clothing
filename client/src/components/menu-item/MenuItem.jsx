import React from "react";
import { withRouter } from "react-router-dom";

import {
  MenuItemContainer,
  BackgroundImageContainer,
  ContentContainer,
  TitleContainer,
  SubtitleContainer
} from "./MenuItemStyles";

const MenuItem = ({ title, imageUrl, size, history, match }) => {
  return (
    <MenuItemContainer
      size={size}
      onClick={() => history.push(`${match.url}shop/${title}`)}
    >
      <BackgroundImageContainer
        className="background-image"
        imageUrl={imageUrl}
      />
      <ContentContainer className="content">
        <TitleContainer>{title}</TitleContainer>
        <SubtitleContainer>Shop now</SubtitleContainer>
      </ContentContainer>
    </MenuItemContainer>
  );
};

export default withRouter(MenuItem);
