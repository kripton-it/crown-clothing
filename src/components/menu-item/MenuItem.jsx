import React from "react";
import { withRouter } from "react-router-dom";

import "./MenuItem.styles.scss";

const MenuItem = ({ title, imageUrl, size, history, match }) => {
  return (
    <div
      className={`menu-item menu-item-${size}`}
      onClick={() => history.push(`${match.url}${title}`)}
    >
      <div
        className="background-image"
        style={{ backgroundImage: `url(${imageUrl})` }}
      />
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">Shop now</span>
      </div>
    </div>
  );
};

export default withRouter(MenuItem);
