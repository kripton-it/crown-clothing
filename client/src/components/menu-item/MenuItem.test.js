import React from "react";
import { shallow } from "enzyme";

import { MenuItem } from "./MenuItem";

describe("MenuItem component", () => {
  let wrapper;
  const mockHistory = {
    push: jest.fn()
  };
  const mockMatch = {
    url: "/"
  };
  const mockProps = {
    title: "hello",
    history: mockHistory,
    match: mockMatch,
    imageUrl: "abc.com",
    size: "small"
  };

  wrapper = shallow(<MenuItem {...mockProps} />);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("should render info:", () => {
    it("title", () => {
      expect(wrapper.find("TitleContainer").text()).toBe("hello");
    });

    it("should render imageUrl as a prop on BackgroundImageContainer", () => {
      expect(wrapper.find("BackgroundImageContainer").prop("imageUrl")).toBe(
        "abc.com"
      );
    });

    it("should render size as a prop on MenuItemContainer", () => {
      expect(wrapper.find("MenuItemContainer").prop("size")).toBe("small");
    });
  });

  describe("should call methods:", () => {
    it("should call history.push when MenuItem is clicked", () => {
      wrapper.find("MenuItemContainer").simulate("click");
      const expectedArgument = `${mockMatch.url}shop/hello`;

      expect(mockHistory.push).toHaveBeenCalledWith(expectedArgument);
    });
  });
});
