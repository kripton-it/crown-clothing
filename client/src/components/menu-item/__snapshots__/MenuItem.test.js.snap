// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component should render 1`] = `ShallowWrapper {}`;
